trackintel/analysis/tracking_quality.py

```diff
@@ -0,0 +1,139 @@
+        # calculate per-user per-day tracking quality
+        quality = df.groupby(["user_id", "day"], as_index=False).apply(_get_tracking_quality_user, granularity)
+    elif granularity == "hour":
+        # split records that span several hours
+        df = _split_overlaps(df, granularity="hour")
+        # get the tracked day relative to the first day
+        start_date = df["started_at"].min().date()
+        df["day"] = df["started_at"].apply(lambda x: (x.date() - start_date).days)
+        # get the hour
+        df["hour"] = df["started_at"].dt.hour
+        # calculate per-user per-hour tracking quality
+        quality = df.groupby(["user_id", "hour"], as_index=False).apply(_get_tracking_quality_user, granularity)
+    else:
+        raise AttributeError(f"granularity unknown. We only support ['all', 'day', 'hour']. You passed {granularity}")
+    return quality
+
+
+def _get_tracking_quality_user(df, granularity="all"):
+    """Tracking quality per-user per-granularity.
+
+    Parameters
+    ----------
+    df : GeoDataFrame (as trackintel datamodels)
+        The source dataframe
+
+    granularity : {"all", "day", "hour"}, default "all"
+        Determines the extent of the tracking. "all" the entire tracking period,
+        "day" a whole day and "hour" a whole hour.
+
+    Returns
+    -------
+    pandas.Series
+        A pandas.Series object containing the tracking quality
+    """
+    tracked_duration = (df["finished_at"] - df["started_at"]).dt.total_seconds().sum()
+    if granularity == "all":
+        # the whole tracking period
+        extent = (df["finished_at"].max() - df["started_at"].min()).total_seconds()
+    elif granularity == "day":
+        # total seconds of a day
+        extent = 60 * 60 * 24 - 1
+    elif granularity == "hour":
+        # total seconds of a day times number of different days
+        extent = (60 * 60 - 1) * len(df["day"].unique())
+        print(df[["started_at", "finished_at"]])
+        print(extent)
+    return pd.Series([tracked_duration / extent], index=["quality"])
+
+
+def _split_overlaps(source, granularity="day"):
+    """Split input df that have a duration of several days or hours.
+
+    Parameters
+    ----------
+    source : GeoDataFrame (as trackintel datamodels)
+        The source to perform the split
+
+    granularity : {'day', 'hour'}, default 'day'
+        The criteria of spliting. "day" splits records that have duration of several
+        days and "hour" splits records that have duration of several hours.
+
+    Returns
+    -------
+    GeoDataFrame (as trackintel datamodels)
+        The GeoDataFrame object after the spliting
+    """
+    df = source.copy()
+    if granularity == "day":
+        change_flag = df["started_at"].dt.date != df["finished_at"].dt.date
+    elif granularity == "hour":
+        change_flag = df["started_at"].dt.hour != df["finished_at"].dt.hour
+    i = 0
+
+    while change_flag.sum() > 0:
+
```

Comment:
Maybe add a one-liner how the algorithm works in general:
Something like: 
Iteratively split away one day/hour from multi day/hour entries until no entry spans over multiple days/hours